### ISSUE TITLE
java.lang.IllegalArgumentException when executing example in Titan documentation

### ISSUE THREAD

**Issue by ccoffey** [thinkaurelius/titan#782]
Version: [Titan 0.5.1 with Hadoop 2](http://s3.thinkaurelius.com/downloads/titan/titan-0.5.1-hadoop2.zip)

Steps the recreate the problem.
gremlin> g = TitanFactory.open('conf/titan-berkeleydb-es.properties')
==>titangraph[berkeleyje:/home/cathalc/Desktop/titan-0.5.1-hadoop2/conf/../db/berkeley]
gremlin> GraphOfTheGodsFactory.load(g)
==>null
gremlin> h = g.V.has('name','hercules').next()
==>v[1536]
gremlin> h.query().labels('battled').direction(OUT).orderBy('time',Order.DESC).limit(10).vertices()._().name
==>cerberus
==>hydra
==>nemean
java.lang.IllegalArgumentException
Display stack trace? [yN] y
java.lang.IllegalArgumentException
        at com.google.common.base.Preconditions.checkArgument(Preconditions.java:77)
        at com.thinkaurelius.titan.graphdb.transaction.StandardTitanTx$VertexConstructor.get(StandardTitanTx.java:419)
        at com.thinkaurelius.titan.graphdb.transaction.StandardTitanTx$VertexConstructor.get(StandardTitanTx.java:408)
        at com.thinkaurelius.titan.graphdb.transaction.vertexcache.GuavaVertexCache.get(GuavaVertexCache.java:65)
        at com.thinkaurelius.titan.graphdb.transaction.StandardTitanTx.getInternalVertex(StandardTitanTx.java:405)
        at com.thinkaurelius.titan.graphdb.query.vertex.VertexLongList.get(VertexLongList.java:53)
        at com.thinkaurelius.titan.graphdb.query.vertex.VertexLongList$1.next(VertexLongList.java:124)
        at com.thinkaurelius.titan.graphdb.query.vertex.VertexLongList$1.next(VertexLongList.java:112)
        at com.tinkerpop.pipes.util.iterators.HistoryIterator.next(HistoryIterator.java:25)
        at com.tinkerpop.pipes.IdentityPipe.processNextStart(IdentityPipe.java:19)
        at com.tinkerpop.pipes.AbstractPipe.next(AbstractPipe.java:89)
        at com.tinkerpop.pipes.transform.PropertyPipe.processNextStart(PropertyPipe.java:29)
        at com.tinkerpop.pipes.AbstractPipe.hasNext(AbstractPipe.java:98)
        at com.tinkerpop.pipes.util.Pipeline.hasNext(Pipeline.java:105)
        at com.tinkerpop.pipes.transform.ToStringPipe.processNextStart(ToStringPipe.java:24)
        at com.tinkerpop.pipes.transform.ToStringPipe.processNextStart(ToStringPipe.java:16)
        at com.tinkerpop.pipes.AbstractPipe.hasNext(AbstractPipe.java:98)
        at com.thinkaurelius.titan.hadoop.tinkerpop.gremlin.ResultHookClosure.call(ResultHookClosure.java:52)
        at groovy.lang.Closure.call(Closure.java:428)
        at sun.reflect.GeneratedMethodAccessor22.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:483)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSite.invoke(PogoMetaMethodSite.java:231)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.call(PogoMetaMethodSite.java:64)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
        at org.codehaus.groovy.tools.shell.Groovysh.setLastResult(Groovysh.groovy:324)
        at org.codehaus.groovy.tools.shell.Groovysh.this$3$setLastResult(Groovysh.groovy)
        at sun.reflect.GeneratedMethodAccessor21.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:483)
        at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
        at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:233)
        at groovy.lang.MetaClassImpl.setProperty(MetaClassImpl.java:2416)
        at groovy.lang.MetaClassImpl.setProperty(MetaClassImpl.java:3347)
        at org.codehaus.groovy.tools.shell.Shell.setProperty(Shell.groovy)
        at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.setGroovyObjectProperty(ScriptBytecodeAdapter.java:528)
        at org.codehaus.groovy.tools.shell.Groovysh.execute(Groovysh.groovy:152)
        at org.codehaus.groovy.tools.shell.Shell.leftShift(Shell.groovy:114)
        at org.codehaus.groovy.tools.shell.Shell$leftShift$0.call(Unknown Source)
        at org.codehaus.groovy.tools.shell.ShellRunner.work(ShellRunner.groovy:88)
        at org.codehaus.groovy.tools.shell.InteractiveShellRunner.super$2$work(InteractiveShellRunner.groovy)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:483)
        at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
        at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:233)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1079)
        at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.invokeMethodOnSuperN(ScriptBytecodeAdapter.java:128)
        at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.invokeMethodOnSuper0(ScriptBytecodeAdapter.java:148)
        at org.codehaus.groovy.tools.shell.InteractiveShellRunner.work(InteractiveShellRunner.groovy:100)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:483)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:272)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:52)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:137)
        at org.codehaus.groovy.tools.shell.ShellRunner.run(ShellRunner.groovy:57)
        at org.codehaus.groovy.tools.shell.InteractiveShellRunner.super$2$run(InteractiveShellRunner.groovy)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:483)
        at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
        at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:233)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1079)
        at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.invokeMethodOnSuperN(ScriptBytecodeAdapter.java:128)
        at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.invokeMethodOnSuper0(ScriptBytecodeAdapter.java:148)
        at org.codehaus.groovy.tools.shell.InteractiveShellRunner.run(InteractiveShellRunner.groovy:66)
        at com.thinkaurelius.titan.hadoop.tinkerpop.gremlin.Console.<init>(Console.java:61)
        at com.thinkaurelius.titan.hadoop.tinkerpop.gremlin.Console.<init>(Console.java:70)
        at com.thinkaurelius.titan.hadoop.tinkerpop.gremlin.Console.main(Console.java:75)

**Comment by spmallette** [comment]
Seems the workaround is to iterate to List first:

``` text
gremlin> h.query().labels('battled').direction(OUT).orderBy('time',Order.DESC).limit(10).vertices().collect{it}._().name
==>cerberus
==>hydra
==>nemean
```

though i don't know why.

**Comment by mbroecheler** [comment]
It seems we introduced a bug in VertexLongList in the latest refactoring without noticing (i.e. no unit test coverage) - it will continuing iterating even though there are no more elements.

**Comment by ccoffey** [comment]
Thanks Matthias, shall I create a github issue or is this already being
tracked?

**Comment by mbroecheler** [comment]
This is a github issue ;-)

**Comment by ccoffey** [comment]
Wow, I think I need to drink less coffee. I'm opening github issues unconsciously ;-)

**Comment by mbroecheler** [comment]
If you are drinking Irish coffee, I'd agree with that.

**Comment by spmallette** [comment]
:beers:

**Comment by ccoffey** [comment]
:-)

**Comment by dalaro** [comment]
@mbroecheler I don't think this is a VertexLongList issue.  I attached a debugger and set breakpoints in VertexLongList's `hasNext` and `next` methods.  

@ccoffey's query never trips the `hasNext` breakpoint.  At first I thought was a problem with my debugger.  I compared with @spmallette's workaround query.  That trips `hasNext` and `next` in alternation, exactly as one would expect.  So  `VertexLongList.hasNext` seems to work, but it's not being invoked in the failure scenario.
